PEPRest/src/main/java/it/cnr/iit/peprest/PEPRest.java

```diff
@@ -92,9 +94,15 @@ public UCSProxy getUCSInterface() {
     }
 
     public String tryAccess() {
-        RequestWrapper request = RequestWrapper.build( FileUtility.readFileAbsPath( pep.getRequestPath() ) );
-        PolicyWrapper policy = PolicyWrapper.build( FileUtility.readFileAbsPath( pep.getPolicyPath() ) );
         log.log( Level.INFO, "TryAccess at {0} ", System.currentTimeMillis() );
+        PolicyWrapper policy;
+        RequestWrapper request;
+        try {
+            policy = PolicyWrapper.build( FileUtility.readFileAbsPath( pep.getPolicyPath() ) );
+            request = RequestWrapper.build( FileUtility.readFileAbsPath( pep.getRequestPath() ) );
+        } catch( PolicyException | RequestException e ) {
+            return INVALID_MESSAGE_ID;
```

Comment:
returning a message indicating the message id is invalid is confusing becsue either policy or request exception has occured.

Reply:
Should tryAccess throw an exception?
Before it was possibly crashing since request wrapper could have been null.

Reply:
May be you can return a message indicating that an exception in policy or request has occured

Reply:
oh ok, this means I can actually give a good value to that INVALID_MESASSAGE_ID ?

Reply:
I would actually change the return of the tryAccess. why do we need the messageId outside? maybe boolean is sufficient.

Reply:
We need it to check the status. Look at the test cases for message Status

Reply:
Ok let's recap.
Currently this function in the PEP is used to return the messageId of the sent message.
This is not actually used for something important.

If an error occurs we either:
1) launch an exception
2) return an invalid messageId
3) change the return to something else.

do you mean to return a PEP_STATUS status value?